api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -226,6 +243,70 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
+    JSONObject notebookFile = new JSONObject()
+        .put("cells", new JSONArray()
+            .put(new JSONObject()
+                .put("cell_type", "code")
+                .put("metadata", new JSONObject())
+                .put("execution_count", JSONObject.NULL)
+                .put("outputs", new JSONArray())
+                .put("source", new JSONArray()
+                  .put(queriesAsStrings))))
+        .put("metadata", new JSONObject())
+        .put("nbformat", 4)
+        .put("nbformat_minor", 2);
+
+    WorkspaceResponse workspace = fireCloudService.getWorkspace(workspaceNamespace, workspaceId);
+
+    cloudStorageService.writeFile(workspace.getWorkspace().getBucketName(),
+        "notebooks/" + dataSetExportRequest.getNotebookName() + ".ipynb", notebookFile.toString().getBytes(StandardCharsets.UTF_8));
+
+
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  private String convertQueryToString(DataSetQuery query, String prefix) {
+    String namespace = prefix.toLowerCase().replaceAll(" ", "_") + "_" + query.getDomain().toString().toLowerCase() + "_";
+    String sqlSection = namespace + "sql = \"\"\"" + query.getQuery() + "\"\"\"";
+
+    String namedParamsSection = namespace +
+        "query_config = {\n" +
+        "  \'query\': {\n" +
+        "\'parameterMode\': \'NAMED\',\n" +
+        "\'queryParameters\': [\n" +
+        query.getNamedParameters().stream().map(namedParameterEntry -> convertNamedParameterToString(namedParameterEntry) + "\n").collect(Collectors.joining("")) +
+        "    ]\n" +
+        "  }\n" +
+        "}\n\n";
+
+    String dataFrameSection = namespace + "df = pandas.read_gbq(" + namespace + "sql, dialect=\"standard\", configuration=" + namespace + "query_config)";
+
+    return sqlSection + "\n\n" + namedParamsSection + "\n\n" + dataFrameSection;
+  }
+
+  private String convertNamedParameterToString(NamedParameterEntry namedParameterEntry) {
+    if (namedParameterEntry.getValue() != null) {
+      boolean isArrayParameter = !(namedParameterEntry.getValue().getParameterValue() instanceof String);
+      return "      {\n" +
+          "        'name': \"" + namedParameterEntry.getValue().getName() + "\",\n" +
+          "        'parameterType': {'type': \"" + namedParameterEntry.getValue().getParameterType() + "\"" +
+          (isArrayParameter ? ",'arrayType': {'type': \"" + namedParameterEntry.getValue().getArrayType() + "\"}," : "") + "},\n" +
+          "        \'parameterValue\': {" + (isArrayParameter ? "\'arrayValues\': [" + (
+              (List<NamedParameterValue>) namedParameterEntry.getValue().getParameterValue())
+          .stream().map(namedParameterValue -> "{\'value\': " + namedParameterValue.getParameterValue() + "}") + "]" :
+          "'value': \"" + namedParameterEntry.getValue().getParameterValue() + "\"") + "}\n" +
+          "      },";
+    } else {
+      return "";
```

Comment:
nit: we could also do like if (namedParameterEntry.getValue() == null) return "" and then the rest of the code.